app/models/piece.rb

```diff
@@ -3,56 +3,41 @@ class Piece < ActiveRecord::Base
   belongs_to :user
   belongs_to :game
 
-def obstructed_move?(x, y)
-  current_row = self.row_position
-  current_col = self.col_position
-
-  # check if piece is blocking the path in positive vertical direction
-  if current_col == y && x > current_row
-    for i in (current_row+1...x) do
-      if self.game.piece_exists?(i, y)
-        return true
+  # is_obstructed? determines if there are obstructions between two squares. An obstruction is “something between point A and point B”.
+  # Raise an Error Message if Not diagnal, horizontal, or vertical. (Sample is Knight)
+  # Return true if there is a piece found between current x/y to target x/y else return false
+  # Raise an error is target is not vertical, horizontal, and diagonal.
+  def is_obstructed?(x, y)
+    current_row = self.row_position
+    current_col = self.col_position
+
+    # check if piece is blocking the path in positive vertical direction
+    if current_col == y && x > current_row
+      for i in (current_row + 1...x)
+        return game.occupied?(i, y)
```

Comment:
check the logic on these.  If I have a piece at (4, 4) and moving to (1, 4) I want to look for obstructions at (3, 4), (2, 4) only.  (4, 4) is occupied by the piece that is moving.   (1, 4) will be handled differently.  Any thoughts why?